pkg/stream/message.go

```diff
@@ -38,3 +38,9 @@ func (m *Message) UnmarshalFromBytes(data []byte) error {
 func (m *Message) UnmarshalFromString(data string) error {
 	return m.UnmarshalFromBytes([]byte(data))
 }
+
+func (m *Message) WithGzipCompression() *Message {
```

Comment:
I'm not so sure about this one. It misleads a little bit into thinking that calling this would compress the message..  
Instead it's just setting an attribute..

Reply:
good point. I guess it either should be changed to allow anyone to set any message attribute or to also perform the compression. Will have to rethink this.